app/views/comments/_comment_form.html.erb

```diff
@@ -2,115 +2,120 @@
 <% if !commentable && @commentable %>
   <% commentable = @commentable %>
 <% end %>
-<div class="post comment" id="comment_form_for_<%= commentable.id %>">
-  <%= form_for value_for_comment_form(commentable, comment), :remote => !comment.new_record?, :html => {:id => "comment_for_#{commentable.id}"} do |f| %>
-    <fieldset>
-      <legend><%= ts("Post Comment") %></legend>
 
-      <% # here come the hacks (hidden fields to transmit various info to the create action) %>
-      <% if commentable.is_a?(Tag) %>
-        <%= hidden_field_tag :tag_id, commentable.name %>
-      <% end %>
+<% if logged_in_as_admin? %>
+  <% # Hide form for admins %>
+<% else %>
```

Comment:
Would `unless logged_in_as_admin?` work here, rather than including an empty `if` for admins?

Reply:
Maybe it'd be better to move the conditional out of this file? Looking at the places where `comments/comment_form` is rendered, most of those should be inaccessible to admins with a few other (probably necessary) changes like:
1. Removing the Reply button for admins by modifying `can_reply_to_comment?`.
2. Preventing admins from accessing the `CommentsController#new` action.
3. Preventing admins from accessing the `CommentsController#add_comment_reply` action.

Since admins can't access user inboxes, and can't edit comments, that would leave only the call in `comments/commentable` to modify.

Reply:
@tickinginstant 

Unsure about such an approach in the midterm. We know the various ways to get to the comment form as of now. But if a new one is added later on, there's a good chance for the "not admin check" to be forgotten and the issue to return.

Reply:
IMO it's not a big deal if `comments/comment_form` is added to a new place later on, because the actual creation of comments still goes through the `create` action, which will be blocked. So it'll be a visual bug at most (and a very obvious visual bug, which is generally pretty easy to find/fix/test).

If you stick to wrapping `comments/comment_form` in a giant if statement, and don't add checks elsewhere, you're going to leave behind empty `<div>`s in places like this:
https://github.com/otwcode/otwarchive/blob/23ffdc103362f4c3c61167a31066d1fe87ed7167/app/views/comments/_commentable.html.erb#L109-L113
Which feels a bit messy to me. And if you do have checks elsewhere, you don't need them in `comments/comment_form` as well.

Reply:
Just modifying the CommentsController isn't enough to prevent the form from being displayed in all cases (this controller is not called when the template is just rendered in another template). I do agree however it makes more sense to perform the check in the `comments/_commentable.html.erb`file, alongside all the others, than adding a gigantic wrapper one step later.

Reply:
Thanks for making the changes! Which of the actions/templates hasn't been handled yet?

Reply:
Based on the files `comments/comment_form` is rendered in, I think we should be good?

* `app/views/comments/_commentable.html.erb` => Already updated, so good
* `app/views/comments/_single_comment.html.erb`, `app/views/comments/edit.html.erb`, `app/views/comments/edit.js.erb` => Only for existing comment edition, with a check on ownership, so should be good (see no reason to prevent admins from editing their old comments they created before this PR, on the contrary)
* `app/views/comments/_comment_actions.html.erb`, `app/views/comments/add_comment_reply.js.erb`, `app/views/inbox/reply.js.erb` -> Behind a `can_reply_to_comment?` check, an already updated method, so good
* `app/views/comments/new.html.erb` => Behind the new controller check, so good

Unless I'm missing something here?